Polyfills/WebSocket/Source/WebSocket.cpp

```diff
@@ -165,11 +171,11 @@ namespace Babylon::Polyfills::Internal
 
     void WebSocket::OpenCallback()
     {
-        m_runtimeScheduler([this]() {
+        m_runtimeScheduler([this, cancellationSource{m_cancellationSource}]() {
             m_readyState = ReadyState::Open;
             try
             {
-                if (!m_onopen.IsEmpty())
+                if (!cancellationSource->cancelled() && !m_onopen.IsEmpty())
```

Comment:
This check should be at the beginning of this scope. `m_readyState` might be dead also. I think you should try removing these fixes for WebSocket for now and see if the problem still persists.

Reply:
As said offline, this change is mandatory (at least) for Chakra. I've changed the scope here.